### PR TITLE
[TS] Improve EmailOptions def

### PR DESCRIPTION
**Pull request by pierre-H**
See https://nodemailer.com/message/

### CODE REVIEW

**Comment by pierre-H** [comment]
Note : according to me it's important to have good typescript definitions because we don't have developer documentation.
So, the best way to understand how to use Directus services is to read the code.

**Comment by rijkvanzanten** [comment]
> Note : according to me it's important to have accurate typescript definitions because we don't have developer documentation.
> So, the best way to understand how to use Directus services is to read the code.

I totally agree. Funnily enough though, the type definitions that you altered _were_ correct before, as the mailer is only using those fields:

https://github.com/directus/directus/blob/3b82b3e568bd1e09917f4dff47ce75c0c03d82da/api/src/services/mail/index.ts#L64

🙂  

Do you want to update the mailer to support `cc` and `bcc` @pierre-H? We could also look into passing the full object, and using the original `nodemailer` type here

**Comment by pierre-H** [comment]
You're right @rijkvanzanten !
Thank you for your comment !
I just added `cc` and `bcc` like you suggested.

File: api/src/services/mail/index.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ export class MailService {
 		}
 
 		try {
-			await mailer.sendMail({ to, from, subject, html, text });
+			await mailer.sendMail({ to, cc, bcc from, subject, html, text });
```
Comment (rijkvanzanten) [inline api/src/services/mail/index.ts]:
```suggestion
			await mailer.sendMail({ to, cc, bcc, from, subject, html, text });
```

Comment (rijkvanzanten) [reply api/src/services/mail/index.ts]: nodemailer supports way more things even (https://nodemailer.com/message/). Should we try doing something like

```
sendMail({
   ...nodeMailerOptions,
   to, cc, bcc, from, subject
});
```

so we can allow setting any of the nodemailer options?

Comment (pierre-H) [reply api/src/services/mail/index.ts]: Oh you're right !
I edited the file so now we have all nodemailer options.

**Review by rijkvanzanten — DISMISSED** [review]

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**